### ISSUE TITLE
OpenAPIRenderer raises AttributeError when the response object is not a Document

### ISSUE THREAD

**Issue by unklphil** [marcgibbons/django-rest-swagger#534]
For example, when not logged in and loading the swagger docs, the serializer responds with the **dict**:

```python
{'detail': 'Authentication credentials were not provided.'}
```

This then gets passed on to the OpenAPIRenderer, which expects a `coreapi.document.Document` object, which results in the error:

```
Traceback (most recent call last):
  File ".../site-packages/django/core/handlers/base.py", line 174, in get_response
    response = self.process_exception_by_middleware(e, request)
  File ".../site-packages/django/core/handlers/base.py", line 172, in get_response
    response = response.render()
  File ".../site-packages/django/template/response.py", line 160, in render
    self.content = self.rendered_content
  File ".../site-packages/rest_framework/response.py", line 72, in rendered_content
    ret = renderer.render(self.data, accepted_media_type, context)
  File ".../site-packages/rest_framework_swagger/renderers.py", line 16, in render
    data = self.get_openapi_specification(data)
  File ".../site-packages/rest_framework_swagger/renderers.py", line 29, in get_openapi_specification
    return json.loads(codec.dump(data))
  File ".../site-packages/openapi_codec/__init__.py", line 34, in dump
    data = generate_swagger_object(document)
  File ".../site-packages/openapi_codec/encode.py", line 8, in generate_swagger_object
    parsed_url = urlparse.urlparse(document.url)
AttributeError: 'dict' object has no attribute 'url'
```

**Comment by marcgibbons** [comment]
Hi @unklphil - which version are you using? This was resolved in #522 #523

**Comment by unklphil** [comment]
I was using 2.0.4, I didn't realise 2.0.5 has been released. Can confirm it is fixed in 2.0.5

**Comment by marcgibbons** [comment]
Great :smile: !
